internal/keepers/factory.go

```diff
@@ -127,5 +128,6 @@ func (d *keepersReportingFactory) NewReportingPlugin(c types.ReportingPluginConf
 			offChainCfg.MinConfirmations,
 			d.logger,
 		),
+		reportGasLimit: offChainCfg.GasLimitPerReport,
```

Comment:
Where do we have it on the plugin side? 
I saw one on the chainlink node side. As I understand, we will configure it on the chainlink node side as an off-chain config.

Reply:
there's one which is used here https://github.com/smartcontractkit/chainlink/blob/develop/core/services/relay/evm/ocr2keeper.go#L74 within transmitter (not the plugin)

> we will configure it on the chainlink node side as an off-chain config

offchainConfig is set on the contract so it's a different place than chainlink node config

Reply:
Right, but as I understand, we don't have access to this value from the plugin itself. We would need additionally pass it on the chainlink node side.

Reply:
@infiloop2 you're talking about the `OCR2AutomationGasLimit` in the node config, right? What does this currently do and why is it configurable per node? I think we need gas limits in the offChainConfig because all nodes need to generate reports deterministically. Each node having different upper thresholds would cause reports to not match.

Reply:
@EasterTheBunny yes. Currently this is passed to the transmitter and used as the gas limit for all transmit txs. Agree it is set per node and not optimal as different nodes can technically set that to be different. 

Ideally `OCR2AutomationGasLimit` would also be set through offchainConfig but right now there's no way for the plugin to tell the per report gas limit to the transmitter. That probably requires a libocr change (needs to be dug into)

Reply:
Hmmm. We might need to live with both for now until we can find a way to consolidate them. That just makes setting the off chain config values more complex and confusing.